### PR TITLE
feat: add ads localize

### PR DESCRIPTION
**Pull request by guyhguy25**
Add a localize for Advertisements.
 
## Summary

Added props to change the ima language settings.

### Motivation

I have been asked to make the "skip" and "Read more" in different languages.
I tried to ask in Discord if someone knows how to add it, but I didn't see anyone who asked for this feature, although if someone using this library for production he needs it, so I did it.
I created that for my company so it did not waste my time, if you like it you more than welcome to take it.

### Changes

add prop: adLanguage
add docs for it.
#### Java:
   add ImaSdkFactory & ImaSdkSettings to the google ads api
#### Swift:
   see pr diff.

## Test plan

add adTagUrl and then add adLanguage for example: "fr".
and that's it.

```
<Video
    adTagUrl="https://pubads.g.doubleclick.net/gampad/ads?iu=/21775744923/external/vmap_ad_samples&sz=640x480&cust_params=sample_ar%3Dpremidpostoptimizedpodbumper&ciu_szs=300x250&gdfp_req=1&ad_rule=1&output=vmap&unviewed_position_start=1&env=vp&impl=s&cmsid=496&vid=short_onecue&correlator="
    adLanguage="fr"
    source={{
        uri: "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4",
    }}
/>
```

### CODE REVIEW

**Comment by freeboub** [comment]
Looks good !
2 open points:
- can you document the default value ?
- does it take sense to change language on runtime ? I wonder if this new prop should be moved into the source object ...

**Comment by freeboub** [comment]
And please check why the build without ads is ko ! :)

**Comment by guyhguy25** [comment]
> Looks good ! 2 open points:
> 
> * can you document the default value ?
> * does it take sense to change language on runtime ? I wonder if this new prop should be moved into the source object ...

* I can add the default value, but I am not sure I'm correct on this one.
its seems like ios using the system language, and Android defaults to English.

* no the language should not change on runtime. (its a possibility, your choice :D)

> And please check why the build without ads is ko ! :)
weird, I ain't receiving an error. 

I will add the instance to imaSdkFactory, and the function setImaSdkSettings to imaLoader that's probably it.

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.kt [hunk 1]
```diff
@@ -110,6 +111,16 @@ class ReactExoplayerViewManager(private val config: ReactExoplayerConfig) : View
         videoView.setAdTagUrl(adTagUrl)
     }
 
+    @ReactProp(name = PROP_AD_LANGUAGE)
+    fun setAdLanguage(videoView: ReactExoplayerView, languageString: String?) {
+        if (TextUtils.isEmpty(languageString)) {
+            videoView.setAdLanguage(null) // Maybe "en" default?
```
Comment (moskalakamil) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.kt:117]: I think we should leave default behavior when language is non passed

**Comment by moskalakamil** [comment]
Thanks for PR! I'm just wondering, is it possible to create a 'adConfig' prop with all customization for ads? maybe it's similar logic and then we can pass all of the settings for ads

File: docs/pages/component/ads.md [hunk 2]
```diff
@@ -23,3 +23,16 @@ Example:
 onReceiveAdEvent={event => console.log(event)}
 ...
 ```
+
+### Localization
+To change the language of the IMA SDK, you need to pass `adLanguage` prop to `Video` component. List of supported languages, you can find [here](https://developers.google.com/interactive-media-ads/docs/sdks/android/client-side/localization#locale-codes)
+
+by default ios will be system language and android will be `en`
```
Comment (freeboub) [inline docs/pages/component/ads.md]:
```suggestion
By default, ios will use system language and android will use `en`
```

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.kt (hunk 1 above)
Comment (freeboub) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerViewManager.kt:117]: null will fallback on default behavior, so ok for me

**Comment by freeboub** [comment]
> Thanks for PR! I'm just wondering, is it possible to create a 'adConfig' prop with all customization for ads? maybe it's similar logic and then we can pass all of the settings for ads

I agree, but we should not block this PR For that point.

**Comment by freeboub** [comment]
> * no the language should not change on runtime. (its a possibility, your choice :D)
> 
From IMA doc:  
Once the AdsLoader object has been created, using this setter will have no effect.
I have my answer, thank you !

**Comment by freeboub** [comment]
I just put one proposal for doc, I let you approve and integrate it.
This is good to me !

**Review by freeboub — APPROVED** [review]

**Comment by guyhguy25** [comment]
> > Thanks for PR! I'm just wondering, is it possible to create a 'adConfig' prop with all customization for ads? maybe it's similar logic and then we can pass all of the settings for ads
> 
> I agree, but we should not block this PR For that point.

Hey, I don't mind doing this prop, and move this prop there.
I can open a new PR for it

I already implement that on ios. i need to work on android and I will open pr for it :D

**Comment by freeboub** [comment]
@guyhguy25 up to you, I will wait for @KrzysztofMoch next week to merge it

**Review by KrzysztofMoch — APPROVED** [review]
It is Okay to merge for me

**Comment by freeboub** [comment]
OK, let's merge as is. Don't hesitate to reopen another PR for refactor ! @guyhguy25

**Pull request merged**